addons/survey/static/src/question_page/question_page_one2many_field.js

```diff
@@ -83,10 +86,22 @@ class QuestionPageOneToManyField extends X2ManyField {
             if (params.record) {
                 params.record = record.data[name].records.find(r => r.resId === params.record.resId);
             }
+            if (params.context.question_create) {
+                const limit = this.list.config.limit;
+                const offset = (Math.ceil(this.list.count / limit) - 1) * limit;
+                await this.list.load({ limit: limit, offset: offset });
+            }
```

Comment:
Actually, we this new version, we don't need anymore the override of "onAdd" and with the help of the method "loadLastPage", we can rewrite it:
```javascript
if (params.context.question_create) {
    await this.loadLastPage();
}
```
With those 2 changes based on what you have found, we have a load before and after the save.
Before to get the correct index (I think) and after to load the correct page if we jump to the next.